static/fortran-mode.js

```diff
@@ -0,0 +1,571 @@
+// This is put together from two sources:
+// 1) d-mode.js by Matt Godbolt
+//    Copyright (c) 2017, Matt Godbolt
+//    All rights reserved.
+//
+//    Redistribution and use in source and binary forms, with or without
+//    modification, are permitted provided that the following conditions are met:
+//
+//        * Redistributions of source code must retain the above copyright notice,
+//          this list of conditions and the following disclaimer.
+//        * Redistributions in binary form must reproduce the above copyright
+//          notice, this list of conditions and the following disclaimer in the
+//          documentation and/or other materials provided with the distribution.
+//
+//    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+//    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+//    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+//    ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+//    LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+//    CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+//    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+//    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+//    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+//    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+//    POSSIBILITY OF SUCH DAMAGE.
+// 2) vs.language.fortran by Tom Dunn
```

Comment:
If this file is already on `vs`, shouldn't we have Fortan support out-of-the-box already?

Reply:
To be honest I have no clue whether or not this is in `vs` (I've never seen/used `vs`). Also wouldn't know how to find or check for it. I copied bits from https://github.com/tomedunn/vs.language.fortran and merged it with `d-mode.js` adjusting the rest. And as I initially said, I guess it's far from done.

Reply:
After looking through https://github.com/Microsoft/monaco-languages (The repostory that holds all the different syntaxes for Monaco Editor) it does not seem like Fortan is supported out of the box, so it turns out we do need this file after all.

Now, IANAL, but I don't know how valid this double license header is

Reply:
I'm not a lawyer either. I left both in there since the two source files I used both required to keep the copyright notice and I didn't want to take credit for the lines I copied.

Now, `d-mode.js` is probably not too critical since from this repo? And not too much left of it anyway.

All that's left from `vs.language.fortran` are the lists of keywords, functions, etc. And those lists are language specific. And I'd be tempted to argue that `The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.` frees us from keeping his copyright notice. I'd still give some credit to the original file.